tests/unit/amo/components/TestPage.js

```diff
@@ -240,6 +255,466 @@ describe(__filename, () => {
+        _dispatchClientMetadata({ userAgent: userAgents.firefoxAndroid[0] });
+        render(props);
+
+        expect(screen.queryByText('download Firefox')).not.toBeInTheDocument();
+      });
+
+      it('does not render a download banner if the browser is Firefox for iOS', () => {
+        _dispatchClientMetadata({ userAgent: userAgents.firefoxIOS[0] });
+        render(props);
+
+        expect(screen.queryByText('download Firefox')).not.toBeInTheDocument();
+      });
+    });
+  });
+
+  describe('Tests for Header', () => {
+    it('renders an <h1> when isHomepage is true', () => {
+      render({ isHomePage: true });
+
+      expect(
+        screen.getByRole('heading', { name: 'Firefox Browser Add-ons' }),
+      ).toBeInTheDocument();
+      expect(
+        screen.getByRole('link', { name: 'Firefox Browser Add-ons' }),
+      ).toHaveAttribute('href', '/en-US/android/');
+      expect(screen.getByRole('banner')).not.toHaveClass(
+        'Header--loaded-page-is-anonymous',
+      );
+    });
+
+    it('renders a download banner Banner when isAddonInstallPage is false', () => {
+      _dispatchClientMetadata({ userAgent: userAgents.chrome[0] });
+
+      render({ isAddonInstallPage: false });
+
+      expect(screen.getByText('download Firefox')).toBeInTheDocument();
+    });
+
+    it('does not render a download banner Banner when isAddonInstallPage is true', () => {
+      _dispatchClientMetadata({ userAgent: userAgents.chrome[0] });
+
+      render({ isAddonInstallPage: true });
+
+      expect(screen.queryByText('download Firefox')).not.toBeInTheDocument();
+    });
+
+    it('always renders a link in the header when not on homepage', () => {
+      render({ isHomePage: false });
+
+      expect(
+        screen.queryByRole('heading', { name: 'Firefox Browser Add-ons' }),
+      ).not.toBeInTheDocument();
+      expect(
+        screen.getByRole('link', { name: 'Firefox Browser Add-ons' }),
+      ).toHaveAttribute('href', '/en-US/android/');
+    });
+
+    it('displays `Log in` text when user is not signed in', () => {
+      render();
+
+      expect(screen.getByRole('link', { name: 'Log in' })).toBeInTheDocument();
+      expect(screen.queryByText('My Account')).not.toBeInTheDocument();
+    });
+
+    it('displays a menu and the display name when user is signed in', () => {
+      const displayName = 'King of the Elephants';
+      _dispatchSignInActions({
+        userProps: { display_name: displayName },
+      });
+      render();
+
+      expect(
+        screen.getByRole('button', { name: displayName }),
+      ).toBeInTheDocument();
+      expect(screen.getByText('My Account')).toBeInTheDocument();
+    });
+
+    it('displays the username when user is signed in but has no display name', () => {
+      const username = 'babar';
+      _dispatchSignInActions({ userProps: { username } });
+      render();
+
+      expect(
+        screen.getByRole('button', { name: username }),
+      ).toBeInTheDocument();
+    });
```

Comment:
There is always a `name` in the api response; username isn't a fallback when display_name isn't set; and `display_name` itself should only be used in the user profile edit page/component.  I think we're just testing the incorrect fallback in `createUserAccountResponse` here.

Reply:
I see. It's another invalid test because there is actually no logic for this in the `Header` component at all. I'll remove the test. I suppose we should correct the `createUserAccountResponse` helper as, as you state, it seems to be invalid.